src/mapblock_mesh.h

```diff
@@ -161,17 +164,69 @@ class MapBlockMesh
 /*
 	This is used because CMeshBuffer::append() is very slow
 */
-struct PreMeshBuffer
+class PreMeshBuffer
 {
+private:
+	bool m_enable_shaders;
+	u32 m_vertex_pitch;
+public:
 	TileSpec tile;
 	std::vector<u16> indices;
 	std::vector<video::S3DVertex> vertices;
+	std::vector<video::S3DVertexTangents> tangent_vertices;
+
+	PreMeshBuffer(bool enable_shaders):
+	m_enable_shaders(enable_shaders)
+	{
+		if (m_enable_shaders)
+			m_vertex_pitch = getVertexPitchFromType(video::EVT_TANGENTS);
+		else 
+			m_vertex_pitch = getVertexPitchFromType(video::EVT_STANDARD);
+	}
+
+	~PreMeshBuffer()
+	{
+	}	
+
+	u32 getVertexPitch ()
+	{
+		return m_vertex_pitch;
+	}
+
+	u32 getVertexCount ()
+	{
+		if (m_enable_shaders)
+			return tangent_vertices.size();
+		else 
+			return vertices.size();
+	}
+
+	u8 * getVertices()
+	{
+		if (m_enable_shaders)
+			return (u8 *)&tangent_vertices[0];
+		else 
+			return (u8 *)&vertices[0];
+	}
+
+	u8 * getVertex(u32 index)
```

Comment:
By the way, there's an extra space between the * and the function name.  It's supposed to be:
"void *getVertex(u32 index)"